### DISCUSSION TITLE
API fails to update record

### DISCUSSION THREAD

**Discussion by kollross** [ngoduykhanh/PowerDNS-Admin#1271] [Q&A] [answered]
I'm struggling to understand why I can't modify records with a API key.   

Note:  This is a key generated within PDA (not the key stored in dns.conf, which works).

- I've created an Account under Account Management. 
- I then generated a new API under Key Management (as a User, though Operator and Administrator also appear to have no affect), which shows it belongs to that previous Account.  
- Under the domain (manage domain), I've selected the Account from drop down.   The dashboard indicates the domain is part of the Account that I've created. 

However when I make an API call to update a record, I get the following error, in the server logs.
`[webserver] 43c7f189-826d-4320-b69d-6abd04c41a76 HTTP Request "/api/v1/servers/localhost/zones/test.com.": Authentication by API Key failed`

Output from the command itself shows its returning a 401 Unauthorized.

Like I mentioned, changing the API Key to Operator, and Admin seems to have no affect.   Taking PDA key stored in pdns.conf and using that inside of the API call seems to work. 

Any ideas???

**Comment by jbe-dw** [comment, marked as answer]
Hello,

It looks like you're trying to use PDNS API with PDA Apikey. PDA's keys must be used using PDA's API (whose job is to filter and forward requests to PDNS API).

Example: PDNS API listens to localhost:8080 / PDA is accessible at localhost:80

`curl -X GET "http://localhost/api/v1/servers/localhost/zones/test.com." -H 'Content-Type: application/json' -H 'X-API-Key: PDA_API_KEY'`
or
`curl -X GET "http://localhost:8080/api/v1/servers/localhost/zones/test.com."-H 'Content-Type: application/json' -H 'X-API-Key: PDNS_API_KEY'`

**Reply by jbe-dw** [reply]
Currently the doc for the API is almost non-existant (in docs/API.md) but I'll make an update

**Comment by kollross** [comment]
Thank you very much, this worked.   I missed docs/API.md, I was looking inside of the Wiki, thinking that was the primary doc location.
